src/game/option/option_passport.c

```diff
@@ -140,118 +159,129 @@ static void Option_PassportShutdownText(void)
+        m_PassportStatus.pages[PAGE_1] = g_SavedGamesCount > 0;
+        m_PassportStatus.pages[PAGE_2] = false;
+        m_PassportStatus.pages[PAGE_3] = false;
+        m_PassportStatus.modes[PAGE_1] = PASSPORT_MODE_LOAD_GAME;
+        m_PassportStatus.modes[PAGE_2] = PASSPORT_MODE_UNAVAILABLE;
+        m_PassportStatus.modes[PAGE_3] = PASSPORT_MODE_UNAVAILABLE;
+        Option_PassportInitSaveRequester(PAGE_1);
+        break;
+    case INV_SAVE_MODE:
+        m_PassportStatus.mode = PASSPORT_MODE_SAVE_GAME;
+        m_PassportStatus.pages[PAGE_1] = false;
+        m_PassportStatus.pages[PAGE_2] = true;
+        m_PassportStatus.pages[PAGE_3] = false;
+        m_PassportStatus.modes[PAGE_1] = PASSPORT_MODE_UNAVAILABLE;
+        m_PassportStatus.modes[PAGE_2] = PASSPORT_MODE_SAVE_GAME;
+        m_PassportStatus.modes[PAGE_3] = PASSPORT_MODE_UNAVAILABLE;
+        if (g_CurrentLevel == g_GameFlow.gym_level_num) {
+            m_PassportStatus.mode = PASSPORT_MODE_NEW_GAME;
+            m_PassportStatus.modes[PAGE_2] = PASSPORT_MODE_NEW_GAME;
+        } else if (g_Config.enable_save_crystals) {
+            m_PassportStatus.mode = PASSPORT_MODE_RESTART;
+            m_PassportStatus.modes[PAGE_2] = PASSPORT_MODE_RESTART;
+            m_PassportStatus.pages[PAGE_3] = true;
+            m_PassportStatus.modes[PAGE_3] = PASSPORT_MODE_EXIT_TITLE;
         }
-        m_PassportMode = PASSPORT_MODE_FLIP;
-    } else {
-        g_Input = (INPUT_STATE) { 0 };
-        g_InputDB = (INPUT_STATE) { 0 };
+        Option_PassportInitSaveRequester(PAGE_2);
+        break;
+    case INV_SAVE_CRYSTAL_MODE:
+        m_PassportStatus.mode = PASSPORT_MODE_SAVE_GAME;
+        m_PassportStatus.pages[PAGE_1] = false;
+        m_PassportStatus.pages[PAGE_2] = true;
+        m_PassportStatus.pages[PAGE_3] = false;
+        m_PassportStatus.modes[PAGE_1] = PASSPORT_MODE_UNAVAILABLE;
+        m_PassportStatus.modes[PAGE_2] = PASSPORT_MODE_SAVE_GAME;
+        m_PassportStatus.modes[PAGE_3] = PASSPORT_MODE_UNAVAILABLE;
+        Option_PassportInitSaveRequester(PAGE_2);
+        break;
+    case INV_DEATH_MODE:
+        m_PassportStatus.mode = PASSPORT_MODE_BROWSE;
+        m_PassportStatus.pages[PAGE_1] = g_SavedGamesCount > 0;
+        m_PassportStatus.pages[PAGE_2] = true;
+        m_PassportStatus.pages[PAGE_3] = true;
+        m_PassportStatus.modes[PAGE_1] = PASSPORT_MODE_LOAD_GAME;
+        m_PassportStatus.modes[PAGE_2] = PASSPORT_MODE_RESTART;
+        m_PassportStatus.modes[PAGE_3] = PASSPORT_MODE_EXIT_TITLE;
+        break;
+    default:
+        m_PassportStatus.mode = PASSPORT_MODE_BROWSE;
+        m_PassportStatus.pages[PAGE_1] = g_SavedGamesCount > 0;
+        m_PassportStatus.pages[PAGE_2] = true;
+        m_PassportStatus.pages[PAGE_3] = true;
+        m_PassportStatus.modes[PAGE_1] = PASSPORT_MODE_LOAD_GAME;
+        m_PassportStatus.modes[PAGE_2] = PASSPORT_MODE_NEW_GAME;
+        m_PassportStatus.modes[PAGE_3] = PASSPORT_MODE_EXIT_TITLE;
```

Comment:
This is brilliant. Let's just add blank lines after each `break;` for easier vertical navigation.